mlflow/models/evaluation/base.py

```diff
@@ -1265,6 +1265,7 @@ def _validate_dataset_type_supports_predictions(data, supported_predictions_data
 def _evaluate(
     *,
     model,
+    model_predict_func,
```

Comment:
@jessechancy This will break all existing third-party evaluator plugins, including ones developed by Databricks. Can we construct `model_predict_func` inside the default evaluator instead?

Reply:
+1 that it seems naturally scoped to `_extract_predict_fn` except in the case that a user would pass this explicitly into a DefaultEvaluator. Is there a clear use case around that we are trying to address?

Reply:
I think the `_extract_predict_fn` doesn't cover all cases here, since I see direct usage of `model.predict`